### ISSUE TITLE
Call to EF Core Add-Migration fails with System.IO.FileLoadException: Could not load file or assembly 'Microsoft.EntityFrameworkCore, Version=2.1.4.0. 

### ISSUE THREAD

**Issue by jimfightingchance** [dotnet/efcore#14439]
Call to EF Core Add-Migration fails with System.IO.FileLoadException: Could not load file or assembly 'Microsoft.EntityFrameworkCore, Version=2.1.4.0.  See error below.  This seems to have coincided with upgrading Visual Studio 15.9.5.  Using Microsoft.EntityFrameworkCore.SqlServer 2.1.2.  Not sure if it will help but can't move to 2.2.1 as I am waiting on Azure Functions to support it.

```System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.IO.FileLoadException: Could not load file or assembly 'Microsoft.EntityFrameworkCore, Version=2.1.4.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor..ctor(Object reportHandler, IDictionary args)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.RuntimeType.CreateInstanceImpl(BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes)
   at Microsoft.EntityFrameworkCore.Tools.ReflectionOperationExecutor..ctor(String assembly, String startupAssembly, String projectDir, String dataDirectory, String rootNamespace, String language)
   at Microsoft.EntityFrameworkCore.Tools.Commands.ProjectCommandBase.CreateExecutor()
   at Microsoft.EntityFrameworkCore.Tools.Commands.MigrationsAddCommand.Execute()
   at Microsoft.DotNet.Cli.CommandLine.CommandLineApplication.Execute(String[] args)
   at Microsoft.EntityFrameworkCore.Tools.Program.Main(String[] args)
```

### Steps to reproduce
1.  Open package manager console.  
2.  Add-Migration TestMigration

### Further technical details
EF Core version: 2.1.2
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: Windows 10
IDE: (e.g. Visual Studio 2017 15.9.5)

**Comment by jculverwell** [comment]
Couple of people also mentioned this here https://blogs.msdn.microsoft.com/dotnet/2019/01/08/net-core-january-2019-update/  .  Not sure what the resolution is?

**Comment by ajcvickers** [comment]
@jimfightingchance This looks like it might be a duplicate of https://github.com/aspnet/AspNetCore/issues/3503

If not, please post a small project/solution that reproduces the error.

**Comment by jculverwell** [comment]
To confirm https://github.com/aspnet/AspNetCore/issues/3503 workaround 2 fixed the issue.

I needed to upgrade Microsoft.AspNetCore.App PackageReference to the latest 2.1.x version by adding the Version attribute.

Before:

<PackageReference Include="Microsoft.AspNetCore.App" />

After

<PackageReference Include="Microsoft.AspNetCore.App" Version="2.1.4" />

**Comment by jculverwell** [comment]
Sry code didnt come through on last message

Before:

`<PackageReference Include="Microsoft.AspNetCore.App" />`

After:

`<PackageReference Include="Microsoft.AspNetCore.App" Version="2.1.4" />`
